rfc/kubernetes-operator.md

```diff
@@ -0,0 +1,89 @@
+# Kubernetes operator for Spinnaker
+
+| | |
+|-|-|
+| **Status**     | **Proposed** |
+| **RFC #**      | https://github.com/spinnaker/community/pull/41 |
+| **Author(s)**  | Nicolas Cohen (`@ncknt`) |
+| **SIG / WG**   | Kubernetes SIG |
+
+## Overview
+
+Halyard is great at offering a simple way to configure Spinnaker services but is not easy to operationalize. We've come up with an MVP of a Kubernetes operator for Spinnaker that allows administrators to manage Spinnaker with kubectl (or whatever Kubernetes tool).
+
+We're proposing to move the operator to Spinnaker's GitHub: https://docs.google.com/document/d/1Fs-xK7sOZ4YLSw1ae_cQ_rsB0kMNon9AfYli2a5Udhs/edit?usp=sharing
```

Comment:
What are the success criteria for this new project, and plans for ongoing maintenance?  Will this initially be labeled 'alpha' and have a plan to deprecate if it does not gain traction, or are we effectively promising ongoing support for this from day 1?  Is there a SIG or working group that will own the maintenance of this?  (I see the Kubernetes SIG is listed above, but I think there should be a more precise and specific group responsible for maintenance of this at least for the near future.)

Reply:
I'm not sure how to measure success yet but we're really after installing Spinnaker on Kubernetes faster, will less issues. We want to appeal to users who are after a Kubernetes native tool and feel this is a good first step. We'll track usage and success rate for our customers but there's no telemetry baked in.

It's currently alpha and the CRD is subject to change. We do plan to support and improve the operator (we have engineers actively working on it) and hopefully get feedback and help from the community. We created a `kubernetes-operator` Slack channel and are open to making a new SIG if the Kubernetes SIG is not the right group.

Reply:
I don't see why we couldn't form a WG under the Kubernetes SIG for this project specifically.

Reply:
Yeah, I think a WG under the Kubernetes SIG would be a good solution; mostly I want to make sure there's a direct owner of this for the forseeable future and maintenance of this doesn't immediately fall to the Kubernetes SIG which already has a fair amount of work on its plate.

Reply:
Another possibility is a SIG install to deal with all installation matters. Regardless of how we organize it, I'm happy to take ownership of it.